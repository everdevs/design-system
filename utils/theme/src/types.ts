import { Token, ColorObject } from "@evernest/tokens";

export interface Palette {
	red: Token<ColorObject>;
	green: Token<ColorObject>;
	black: Token<ColorObject>;
	white: Token<ColorObject>;
	grey: Token<ColorObject>;
}

export interface Colors {
	primary: Token<ColorObject>;
	secondary: Token<ColorObject>;
	tertiary: Token<ColorObject>;
	lightText: Token<ColorObject>;
	darkText: Token<ColorObject>;
}

export interface Space {
	xxs: number;
	xs: number;
	s: number;
	m: number;
	l: number;
	xl: number;
	xxl: number;
}

export interface Spaces {
	xs: Space;
	s: Space;
	m: Space;
	l: Space;
	xl: Space;
}

export interface Breakpoints {
	xs: number;
	s: number;
	m: number;
	l: number;
	xl: number;
}

export interface Mediaqueries {
	xs: string;
	s: string;
	m: string;
	l: string;
	xl: string;
}

export interface Theme {
	palette: Palette;
	colors: Colors;
	spaces: Spaces;
	breakpoints: Breakpoints;
	mq: Mediaqueries;
}
