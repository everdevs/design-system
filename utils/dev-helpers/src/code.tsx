import React from "react";

export interface CodeProps extends React.HTMLAttributes<HTMLPreElement> {
	code: string;
}

// export const Code: React.FC<CodeProps> = ({ className, code }) => {
// 	return (
// 		<SyntaxHighlighter className={className} language="javascript" style={docco}>
// 			{code}
// 		</SyntaxHighlighter>
// 	);
// };

export const Code: React.FC<CodeProps> = ({ className, code }) => {
	return (
		<pre className={className}>
			<code>{code}</code>
		</pre>
	);
};
