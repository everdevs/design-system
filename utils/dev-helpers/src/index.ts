export * from "./storybook";
export * from "./code";
