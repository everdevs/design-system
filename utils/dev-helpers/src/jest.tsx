import React from "react";
import renderer from "react-test-renderer";

export function createTree(component: React.ReactNode): JSON {
	return renderer.create(component).toJSON();
}

export function withTheme<T = {}>(Component, theme) {
	const ThemedComponent: React.FC<T> = props => <Component {...props} theme={theme} />;
	return ThemedComponent;
}
