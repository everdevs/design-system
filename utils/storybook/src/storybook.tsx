import React from "react";
import { css, Global } from "@emotion/core";
import styled from "@emotion/styled";

export const globalStyle = css`
	body {
		margin: 0;
	}
	*,
	*::before,
	*::after {
		box-sizing: border-box;
	}
`;

export const StyledWrapper = styled.div`
	padding: 1rem;
`;

export const Wrapper: React.FC = ({ children }) => (
	<StyledWrapper>
		<Global styles={globalStyle} />
		{children}
	</StyledWrapper>
);
