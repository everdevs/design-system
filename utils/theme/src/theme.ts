import {
	colorBlack as black,
	colorGreen as green,
	colorGrey as grey,
	colorRed as red,
	colorWhite as white,
} from "@evernest/tokens";
import { Breakpoints, Colors, Mediaqueries, Palette, Spaces, Theme } from "./types";

const palette: Palette = {
	red,
	green,
	black,
	white,
	grey,
};

const colors: Colors = {
	primary: black,
	secondary: white,
	tertiary: grey,
	lightText: white,
	darkText: black,
};

export const spaces: Spaces = {
	xs: {
		xxs: 8,
		xs: 16,
		s: 24,
		m: 32,
		l: 40,
		xl: 48,
		xxl: 80,
	},
	s: {
		xxs: 8,
		xs: 16,
		s: 24,
		m: 32,
		l: 40,
		xl: 48,
		xxl: 80,
	},
	m: {
		xxs: 8,
		xs: 16,
		s: 24,
		m: 32,
		l: 40,
		xl: 48,
		xxl: 80,
	},
	l: {
		xxs: 8,
		xs: 16,
		s: 24,
		m: 32,
		l: 40,
		xl: 48,
		xxl: 80,
	},
	xl: {
		xxs: 8,
		xs: 16,
		s: 24,
		m: 32,
		l: 40,
		xl: 48,
		xxl: 80,
	},
};

export const breakpoints: Breakpoints = {
	xs: 0,
	s: 320,
	m: 640,
	l: 960,
	xl: 1280,
};

export const mq: Mediaqueries = Object.entries(breakpoints).reduce(
	(previousValue, [key, value]: [keyof Breakpoints, number]) => ({
		...previousValue,
		[key]: `(min-width: ${value}px)`,
	}),
	{}
) as Mediaqueries;

export const theme: Theme = {
	palette,
	colors,
	spaces,
	breakpoints,
	mq,
};
