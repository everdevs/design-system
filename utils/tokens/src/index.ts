import * as button from "./tokens/button";
import * as color from "./tokens/color";
import * as font from "./tokens/font";
import * as radius from "./tokens/radius";
import * as shadow from "./tokens/shadow";
import * as space from "./tokens/space";

export * from "./types";
export * from "./utils";

export { button, color, font, radius, shadow, space };
