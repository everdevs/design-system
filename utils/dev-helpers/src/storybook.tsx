import { css, Global } from "@emotion/core";
import { PropsWithTheme, theme } from "@evernest/theme";
import styled from "@emotion/styled";
import { ThemeProvider } from "emotion-theming";
import React from "react";

export function random(max: number, min: number = 0): number {
	return Math.floor(Math.random() * max) + min;
}

export function randomColor(): string {
	return `hsla(${random(359)},50%,50%,0.5)`;
}

export const range = (n: number, start = 1): number[] =>
	new Array(n).fill(Boolean).map((x, i) => i + start);

export const globalStyle = css`
	body {
		margin: 0;
		min-height: 100vh;
	}
	*,
	*::before,
	*::after {
		box-sizing: border-box;
	}
	.sb-show-main {
		padding: 0 !important;
	}
`;

export const StyledWrapper = styled.div<PropsWithTheme>`
	${({ theme: { mq, layout, spaces } }) => css`
		--spacing-xxxs: ${spaces.xxxs}px;
		--spacing-xxs: ${spaces.xxs}px;
		--spacing-xs: ${spaces.xs}px;
		--spacing-s: ${spaces.s}px;
		--spacing-m: ${spaces.m}px;
		--spacing-l: ${spaces.l}px;
		--spacing-xl: ${spaces.xl}px;
		--spacing-xxl: ${spaces.xxl}px;
		--spacing-xxxl: ${spaces.xxxl}px;
		--header-height: ${layout.header.height.s}px;
		@media ${mq.m} {
			--header-height: ${layout.header.height.m}px;
		}
		@media ${mq.l} {
			--header-height: ${layout.header.height.l}px;
		}
	`};
`;

export const Wrapper: React.FC = ({ children }) => (
	<ThemeProvider theme={theme}>
		<StyledWrapper>
			<Global styles={globalStyle} />
			{children}
		</StyledWrapper>
	</ThemeProvider>
);

/* istanbul ignore next */
export const withWrapper = () => storyFn => <Wrapper>{storyFn()}</Wrapper>;

export const ColorBox = styled.div<{
	background?: string;
	color?: string;
	margin?: string;
	height?: string;
}>`
	width: 100%;
	display: inherit;
	box-shadow: 0 2px 3px rgba(0, 0, 0, 0.3);
	border-radius: 2px;
	height: inherit;
	${({ background, color, height }) => css`
		min-height: ${height};
		color: ${color};
		background: ${background};
	`};
	${({ margin }) =>
		margin &&
		css`
			margin: ${margin} 0;
		`};
`;

ColorBox.defaultProps = {
	background: "black",
	color: "white",
	height: "2em",
};
