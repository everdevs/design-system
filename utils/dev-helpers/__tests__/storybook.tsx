import React from "react";
import { mount } from "enzyme";
import { ColorBox, createTree, random, randomColor, range, Wrapper } from "../src";

test("random generates an integer", () => {
	const r = random(10);
	const int = Math.round(r);
	expect(r).toEqual(int);
});

test("random has a max value", () => {
	const max = 10;
	const r = random(max);
	expect(r).toBeLessThanOrEqual(max);
});

test("random has a min value", () => {
	const max = 10;
	const min = 5;
	const r = random(max, min);
	expect(r).toBeGreaterThanOrEqual(min);
});

test("range generates an array", () => {
	const length = 10;
	const arr = range(length);
	expect(arr.length).toEqual(length);
	expect(Array.isArray(arr)).toEqual(true);
});

test("randomColor generates a random color", () => {
	const color = randomColor();
	expect(color).toMatch(/hsla\(\d{0,3},50%,50%,0\.5\)/);
});

test("Wrapper renders children", () => {
	const text = "Hello Button";
	const wrapper = mount(<Wrapper>{text}</Wrapper>);
	expect(wrapper.text()).toEqual(text);
});

test("Wrapper renders correctly", () => {
	const tree = createTree(<Wrapper />);
	expect(tree).toMatchSnapshot();
});

test("ColorBox can change the background", () => {
	const tree = createTree(<ColorBox background="red" />);
	expect(tree).toMatchSnapshot();
});

test("ColorBox can change the color", () => {
	const tree = createTree(<ColorBox color="red" />);
	expect(tree).toMatchSnapshot();
});

test("ColorBox can change the height", () => {
	const tree = createTree(<ColorBox height="10px" />);
	expect(tree).toMatchSnapshot();
});

test("ColorBox can change have top/bottom margins", () => {
	const tree = createTree(<ColorBox margin="10px" />);
	expect(tree).toMatchSnapshot();
});

test("ColorBox renders children", () => {
	const text = "Hello Button";
	const wrapper = mount(<ColorBox>{text}</ColorBox>);
	expect(wrapper.text()).toEqual(text);
});

test("ColorBox allows setting classname", () => {
	const className = "test";
	const wrapper = mount(<ColorBox className={className} />);
	expect(wrapper).toHaveClassName(className);
});
