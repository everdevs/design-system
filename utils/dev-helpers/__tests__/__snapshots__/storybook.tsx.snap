// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorBox can change have top/bottom margins 1`] = `
<div
  className="css-1he52jg"
  color="white"
  height="2em"
/>
`;

exports[`ColorBox can change the background 1`] = `
<div
  className="css-uzy64j"
  color="white"
  height="2em"
/>
`;

exports[`ColorBox can change the color 1`] = `
<div
  className="css-198z4ts"
  color="red"
  height="2em"
/>
`;

exports[`ColorBox can change the height 1`] = `
<div
  className="css-bt35fh"
  color="white"
  height="10px"
/>
`;

exports[`Wrapper renders correctly 1`] = `
<div
  className="css-1mo6rv1"
/>
`;
